docs/proposal-006-build-improvements.md

```diff
@@ -0,0 +1,55 @@
+# Design Proposal #006: Build Improvements
+
+## Current State
+
+At the start of the project we've picked CircleCI as it appears to be a default choice at Weaveworks.
+
+The original solution involved:
+- running `docker build` to create a build image
+- use build image to run tests produce final image
+
+Publishing build and final images hasn't been automated, but some users requested it.
+
+When we switched [to modules](https://github.com/weaveworks/eksctl/pull/917), we have increased the
+compexity of how build image worked. It's hard for a new person to understand it, e.g. for a new
+contributor.
+
+We currently build images in CI and run tests as part of that, we should only run tests as a primary
+task.
+
+Downside of current setup:
+
+- CircleCI requires delegation of access to GitHub
+  - for releases we use a token
+  - we don't have a way to automate workflows that require pushing to the repo
+- CircleCI cache 
+  - it is not very fast, it turns out pulling image with cached dependencies is as fast
+  - it is very specifig to CircleCI, not portable
+  - we do not have a way to controll it
+- the task of building images should not be mixed with the task of running test
+- images are not being automatically pushed to a registry
+- update of build image is manual
```

Comment:
how does this change in the new proposed solution?

Reply:
The version is determined based on git object hashes, but publishing of the image is not yet automated in the PR, that can be done easily later on.

Reply:
Can you add it to a "Next steps" section in this document?

Reply:
There is such section at the bottom already.